android/src/main/java/com/swmansion/gesturehandler/core/FlingGestureHandler.kt

```diff
@@ -83,13 +75,19 @@ class FlingGestureHandler : GestureHandler<FlingGestureHandler>() {
       velocityTracker.yVelocity.toDouble()
     )
 
+    val alignmentList = arrayOf(
+      compareAlignment(velocityVector, SimpleVector(-1.0, 0.0), DIRECTION_LEFT),
+      compareAlignment(velocityVector, SimpleVector(1.0, 0.0), DIRECTION_RIGHT),
+      compareAlignment(velocityVector, SimpleVector(0.0, -1.0), DIRECTION_UP),
+      compareAlignment(velocityVector, SimpleVector(0.0, 1.0), DIRECTION_DOWN)
+    )
+
     velocityTracker.recycle()
 
-    // hypot may be overkill for this simple function, dot product may be sufficient
     val totalVelocity = hypot(velocityVector.x, velocityVector.y)
 
-    val isAligned = compareSimilarity(directionVector, velocityVector) > 0.75
-    val isFast = totalVelocity > this.minAcceptableDelta
+    val isAligned = alignmentList.reduce { any, element -> any or element }
```

Comment:
```suggestion
    val isAligned = alignmentList.any { it }
```

Same as in `.ts`, just different function.